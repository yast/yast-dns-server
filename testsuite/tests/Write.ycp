/**
 * File:	Write.ycp
 * Package:	Configuration of dns-server
 * Summary:	Testsuite for writting configuration
 * Authors:	Jiri Srain <jsrain@suse.cz>, Lukas Ocilka <locilka@suse.cz>
 * Copyright:	Copyright 2004, Novell, Inc.  All rights reserved.
 *
 * $Id$
 *
 * Testsuite for writting configuration for dns-server
 */

{
    include "testsuite.ycp";
    // testedfiles: DnsServer.pm DNSD.pm
    map I_READ = $[
	"probe" : $[
	    "architecture" : "i386",
	    "has_apm" : true,
	    "has_pcmcia" : false,
	    "has_smp" : false,
	    "system" : [],
	    "memory" : [],
	    "cpu" : [],
	    "cdrom" : $[
		"manual" : [],
	    ],
	    "floppy" : $[
		"manual" : [],
	    ],
	    "is_uml" : false,
	],
	"product" : $[
	    "features" : $[
		"USE_DESKTOP_SCHEDULER" : "0",
		"IO_SCHEDULER" : "cfg",
		"UI_MODE" : "expert",
		"ENABLE_AUTOLOGIN" : "0",
		"EVMS_CONFIG" : "0",
	    ],
	],
	"sysconfig" : $[
	    "SuSEfirewall2" : $["FW_ALLOW_FW_TRACEROUTE":"yes", "FW_AUTOPROTECT_SERVICES":"no", "FW_DEV_DMZ":"", "FW_DEV_EXT":"eth-id-00:c0:df:22:c6:a8", "FW_DEV_INT":"", "FW_IPSEC_TRUST":"no", "FW_LOG_ACCEPT_ALL":"no", "FW_LOG_ACCEPT_CRIT":"yes", "FW_LOG_DROP_ALL":"no", "FW_LOG_DROP_CRIT":"yes", "FW_MASQUERADE":"no", "FW_MASQ_NETS":"", "FW_PROTECT_FROM_INTERNAL":"yes", "FW_ROUTE":"no", "FW_SERVICES_DMZ_IP":"", "FW_SERVICES_DMZ_TCP":"", "FW_SERVICES_DMZ_UDP":"", "FW_SERVICES_EXT_IP":"", "FW_SERVICES_EXT_RPC":"nlockmgr status nfs nfs_acl mountd ypserv fypxfrd ypbind ypasswdd", "FW_SERVICES_EXT_TCP":"32768 5801 5901 dixie domain hostname microsoft-ds netbios-dgm netbios-ns netbios-ssn nfs ssh sunrpc", "FW_SERVICES_EXT_UDP":"222 bftp domain ipp sunrpc", "FW_SERVICES_INT_IP":"", "FW_SERVICES_INT_TCP":"ddd eee fff 44 55 66", "FW_SERVICES_INT_UDP":"aaa bbb ccc 11 22 33"],
	    "personal-firewall" : $["REJECT_ALL_INCOMING_CONNECTIONS":""],
	    "console" : $[ "CONSOLE_ENCODING" : "utf8" ],
	    "language" : $[
		"RC_LANG" : "en_US.UTF-8",
		"ROOT_USES_LANG" : "ctype",
	    ],
	],
	"target" : $[
	    "yast2" : $[ "lang2iso.ycp" : $[] ],
	    "stat" : $[
	        "atime":1101890288, "ctime":1101890286, "gid":0, "inode":29236,
	        "isblock":false, "ischr":false, "isdir":false, "isfifo":false,
	        "islink":false, "isreg":true, "issock":false,
	        "mtime":1101890286, "nlink":1, "size":804, "uid":0
	    ],
	    "ycp" : $[],
	],
    ];
    map I_WRITE = $[];
    map I_EXEC = $[];

    TESTSUITE_INIT ([I_READ, I_WRITE, I_EXEC], nil);

    import "Progress";
    import "DnsServer";
    import "Mode";

    Mode::SetMode("test");

    boolean progress_orig = Progress::set (false);

    map READ = $[
	"probe" : $[
	    "architecture" : "i386",
	    "has_apm" : true,
	    "has_pcmcia" : false,
	    "has_smp" : false,
	    "system" : [],
	    "memory" : [],
	    "cpu" : [],
	    "cdrom" : $[
		"manual" : [],
	    ],
	    "floppy" : $[
		"manual" : [],
	    ],
	    "is_uml" : false,
	],
	"product" : $[
	    "features" : $[
		"USE_DESKTOP_SCHEDULER" : "0",
		"IO_SCHEDULER" : "cfg",
		"UI_MODE" : "expert",
		"ENABLE_AUTOLOGIN" : "0",
		"EVMS_CONFIG" : "0",
	    ],
	],

	// Runlevel
        "init": $[
            "scripts": $[
                "exists": true,
                "runlevel": $[
                    "named": $[
                        "start": [],
                        "stop":  [],
                        ],
                    ],
                // their contents is not important for ServiceAdjust
                "comment": $[
                    "named": $[],
                    ],
                ],
            ],
	"target" : $[
	    "stat" : $[
	        "atime":1101890288, "ctime":1101890286, "gid":0, "inode":29236,
	        "isblock":false, "ischr":false, "isdir":false, "isfifo":false,
	        "islink":false, "isreg":true, "issock":false,
	        "mtime":1101890286, "nlink":1, "size":804, "uid":0
	    ],
	    "ycp" : $[],
	],
	"dns" : $[
	    "named" : $[
		"section" : $[
		    "options" : "",
		    "zone \"localhost\" in" : "",
		],
		"value" : $[
		    "options" : $[
			"directory" : "\"/var/lib/named\"",
			"notify" : "no",
			"xxx" : "yyy",
		    ],
		    "zone \"localhost\" in" : $[
			"type" : "master",
			"file" : "\"localhost.zone\"",
		    ],
		],
	    ],
	    "zone" : $[
		"TTL":"1W",
		"records":[
		    $[
			"key":"",
			"type":"NS",
			"value":"@"
		    ], $[
			"key":"",
			"type":"A",
			"value":"127.0.0.1"
		    ], $[
			"key" : "localhost2",
			"type" : "A",
			"value" : "127.0.0.2"
		    ],
		],
		"soa":$[
		    "expiry":"6W",
		    "mail":"root",
		    "minimum":"1W",
		    "refresh":"2D",
		    "retry":"4H",
		    "serial":42,
		    "server":"@",
		    "zone":"@"
		]
	    ],
	],
	"sysconfig" : $[
	    "SuSEfirewall2" : $["FW_ALLOW_FW_TRACEROUTE":"yes", "FW_AUTOPROTECT_SERVICES":"no", "FW_DEV_DMZ":"", "FW_DEV_EXT":"eth-id-00:c0:df:22:c6:a8", "FW_DEV_INT":"", "FW_IPSEC_TRUST":"no", "FW_LOG_ACCEPT_ALL":"no", "FW_LOG_ACCEPT_CRIT":"yes", "FW_LOG_DROP_ALL":"no", "FW_LOG_DROP_CRIT":"yes", "FW_MASQUERADE":"no", "FW_MASQ_NETS":"", "FW_PROTECT_FROM_INTERNAL":"yes", "FW_ROUTE":"no", "FW_SERVICES_DMZ_IP":"", "FW_SERVICES_DMZ_TCP":"", "FW_SERVICES_DMZ_UDP":"", "FW_SERVICES_EXT_IP":"", "FW_SERVICES_EXT_RPC":"nlockmgr status nfs nfs_acl mountd ypserv fypxfrd ypbind ypasswdd", "FW_SERVICES_EXT_TCP":"32768 5801 5901 dixie domain hostname microsoft-ds netbios-dgm netbios-ns netbios-ssn nfs ssh sunrpc", "FW_SERVICES_EXT_UDP":"222 bftp domain ipp sunrpc", "FW_SERVICES_INT_IP":"", "FW_SERVICES_INT_TCP":"ddd eee fff 44 55 66", "FW_SERVICES_INT_UDP":"aaa bbb ccc 11 22 33"],
	    "personal-firewall" : $["REJECT_ALL_INCOMING_CONNECTIONS":""],
	    "console" : $[ "CONSOLE_ENCODING" : "utf8" ],
	    "language" : $[
		"RC_LANG" : "en_US.UTF-8",
		"ROOT_USES_LANG" : "ctype",
	    ],
	],
	"target" : $[
	    "yast2" : $[ "lang2iso.ycp" : $[] ],
	    "ycp" : $[],
	    "stat" : $[
	        "atime":1101890288, "ctime":1101890286, "gid":0, "inode":29236,
	        "isblock":false, "ischr":false, "isdir":false, "isfifo":false,
	        "islink":false, "isreg":true, "issock":false,
	        "mtime":1101890286, "nlink":1, "size":804, "uid":0
	    ],
	],
    ];
    map WRITE = $[];
    map EXEC = $[
	"target" : $[
	    "bash_output" : $["exit" : 0, "stdout" : "20030806"],
	],
    ];

    DUMP ("=======================");
    TEST (``(DnsServer::Import ($[
	"chroot":false,
	"options":[
	    $[
		"key" : "directory",
		"value":"\"/var/lib/named\"",
	    ],
	    $[
		"key" : "notify",
		"value":"no"
	    ],
	],
	"start_service":false,
	"zones":[
	    $[
		"file":"example.com",
		"records":[
		    $[
			"key" : "localhost",
			"type" : "A",
			"value" : "127.0.0.1",
		    ],
		    $[
			"key" : "localhost",
			"type" : "NS",
			"value" : "127.0.0.1",
		    ],
		    $[
			"key" : "localhost2",
			"type" : "A",
			"value" : "127.0.0.2",
		    ],
		],
		"soa":$[
		    "expiry":"6W",
		    "mail":"root",
		    "minimum":"1W",
		    "refresh":"2D",
		    "retry":"4H",
		    "serial":42,
		    "server":"@",
		    "zone":"@"
		],
		"ttl":"1W",
		"type":"master",
		"zone":"example.com",
		"modified" : true,
	    ]
	]])), [], 0);
    DUMP ("=======================");
    TEST (``(DnsServer::Write ()), [READ, WRITE, EXEC], 0);
}

