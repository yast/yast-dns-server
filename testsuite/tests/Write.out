Dump	=======================
Return	true
Dump	=======================
Write	.target.ycp "/var/lib/YaST2/yast2-dns-server_reverse-zones" $[] true
Execute	.target.bash_output "stat --format='rights: %a, blocks: %b, size: %s, owner: %u:%g changed: %Z, modifyied: %Y' /etc/named.conf" $["exit":0, "stdout":"20030806"]
Execute	.target.bash_output "mv --force /etc/named.conf /etc/named.conf.yast2-save" $["exit":0, "stdout":"20030806"]
Execute	.target.bash "touch /etc/named.conf" 0
Write	.dns.named.value.acl [] true
Dir	.dns.named.section: ["options", "zone \"localhost\" in"]
Dir	.dns.named.value.options: ["directory", "notify", "xxx"]
Write	.dns.named.value.options."xxx" nil true
Write	.dns.named.value.options."forwarders" nil true
Write	.dns.named.value.options."directory" ["\"/var/lib/named\""] true
Write	.dns.named.value.options."include" ["\"/etc/named.d/forwarders.conf\""] true
Write	.dns.named.value.options."notify" ["no"] true
Write	.sysconfig.network.config.NETCONFIG_DNS_STATIC_SERVERS "" true
Dir	.dns.named.section: ["options", "zone \"localhost\" in"]
Write	.dns.named nil true
Dir	.dns.named.value: ["options", "zone \"localhost\" in"]
Read	.sysconfig.named.NAMED_CONF_INCLUDE_FILES 0
Write	.sysconfig.named.NAMED_CONF_INCLUDE_FILES "" true
Read	.sysconfig.named.NAMED_INITIALIZE_SCRIPTS 0
Write	.sysconfig.named.NAMED_INITIALIZE_SCRIPTS "createNamedConfInclude" true
Write	.sysconfig.named nil true
Write	.dns.named nil true
Write	.sysconfig.named.NAMED_RUN_CHROOTED "yes" true
Write	.sysconfig.named nil true
Write	.sysconfig.network.config.NETCONFIG_DNS_POLICY "0" true
Read	.sysconfig.named.NAMED_INITIALIZE_SCRIPTS 0
Write	.sysconfig.named.NAMED_INITIALIZE_SCRIPTS "" true
Write	.sysconfig.named nil true
Write	.dns.named.value."zone \"example.com\" in".type ["master"] true
Dir	.dns.named.value."zone \"example.com\" in": []
Execute	.target.bash_output "/bin/hostname --fqdn" $["exit":0, "stdout":"20030806"]
Write	.dns.zone ["/var/lib/named/example.com", $["TTL":"1W", "records":[$["key":"localhost", "type":"A", "value":"127.0.0.1"], $["key":"localhost", "type":"NS", "value":"127.0.0.1"], $["key":"localhost2", "type":"A", "value":"127.0.0.2"]], "soa":$["expiry":"6W", "mail":"root", "minimum":"1W", "refresh":"2D", "retry":"4H", "serial":"42", "server":"@", "zone":"@"]]] true
Write	.dns.named.value."zone \"example.com\" in".file ["\"example.com\""] true
Write	.dns.named.value."zone \"example.com\" in".type ["master"] true
Write	.sysconfig.network.config.NETCONFIG_DNS_FORWARDER "bind" true
Write	.sysconfig.network.config nil true
Execute	.target.bash_output "/sbin/netconfig update -m dns" $["exit":0, "stdout":"20030806"]
Write	.target.ycp "/var/lib/YaST2/dns_server" $[] true
Write	.sysconfig.network.config.NETCONFIG_DNS_FORWARDER "resolver" true
Write	.sysconfig.network.config nil true
Execute	.target.bash_output "/sbin/netconfig update -m dns" $["exit":0, "stdout":"20030806"]
Return	true
